src/ORM/Connect/TempDatabase.php

```diff
@@ -68,6 +73,46 @@ public function isUsed()
         return $this->isDBTemp($selected);
     }
 
+    /**
+     * @return bool
+     */
+    public function hasStarted()
+    {
+        return $this->hasStarted;
+    }
+
+    /**
+     * @return bool
+     */
+    public function supportsTransactions()
+    {
+        return static::getConn()->supportsTransactions();
+    }
+
+    /**
+     * Start a transaction for easy rollback after tests
+     */
+    public function startTransaction()
+    {
+        $this->hasStarted = true;
+        if (static::getConn()->supportsTransactions()) {
+            static::getConn()->transactionStart();
+        }
+    }
+
+    /**
+     * Rollback a transaction (or trash all data if the DB doesn't support databases
+     */
+    public function rollbackTransaction()
+    {
+        if (static::getConn()->supportsTransactions()) {
```

Comment:
Why does this not set hasStarted to false if it supports transactions?

Reply:
So `$hasStarted` is really "do we have the fixtures in the DB" and the only time we don't have the fixtures in the DB is when we clear the DB... it's more "have the tests started" rather than "are we in a transaction"

Reply:
clearAllData actually resets hasStarted so we don't really need it in this method at all